solidity/core/libs/Message.sol

```diff
@@ -1,128 +1,112 @@
 // SPDX-License-Identifier: MIT OR Apache-2.0
 pragma solidity >=0.6.11;
 
-import "@summa-tx/memview-sol/contracts/TypedMemView.sol";
-
 import {TypeCasts} from "./TypeCasts.sol";
 
+// if body is 0 bytes, message is only 1 word long
+struct MessageFingerprint {
+    bytes28 sender;
```

Comment:
Using these structs is super neat and is definitely an improvement for gas & our own dev experience, but I think keeping this as 32 bytes to be able to eventually support non-evm chains like solana/near/whathaveyou is more important than the gains this provides us. I think the struct things are nice but I'm also a little uncomfy with relying so much on the EVM struct packing that's more or less done behind the scenes by the evm

Reply:
its the same behavior as `abi.encodePacked` which we already rely on elsewhere
that being said, I agree that we should stick with `bytes32`